### PR TITLE
Add compresslevel to GZipMiddleware

### PR DESCRIPTION
**Pull request by andamound**
Default level = 9

Documentation gzip: https://docs.python.org/3/library/gzip.html
Benchmarks: https://tukaani.org/lzma/benchmarks.html

### CODE REVIEW

File: starlette/middleware/gzip.py [hunk 1]
```diff
@@ -6,29 +6,36 @@
 
 
 class GZipMiddleware:
-    def __init__(self, app: ASGIApp, minimum_size: int = 500) -> None:
+    def __init__(
+        self, app: ASGIApp, minimum_size: int = 500, compresslevel: int = 1
```
Comment (ycd) [inline starlette/middleware/gzip.py]: You should keep it at 9, which is the default value for [GzipFile](https://docs.python.org/3/library/gzip.html#gzip.GzipFile), or this is going to break a lot of existing code.

File: starlette/middleware/gzip.py [hunk 2]
```diff
@@ -6,29 +6,36 @@
...
+                    self.app, self.minimum_size, compresslevel=self.compresslevel
+                )
                 await responder(scope, receive, send)
                 return
         await self.app(scope, receive, send)
 
 
 class GZipResponder:
-    def __init__(self, app: ASGIApp, minimum_size: int) -> None:
+    def __init__(self, app: ASGIApp, minimum_size: int, compresslevel: int = 1) -> None:
```
Comment (ycd) [inline starlette/middleware/gzip.py]: Also this one.

**Review by ycd — CHANGES_REQUESTED** [review]

File: starlette/middleware/gzip.py (hunk 1 above)
Comment (andamound) [reply starlette/middleware/gzip.py]: I agree. Ready

**Review by Kludex — APPROVED** [review]

**Comment by Kludex** [comment]
@andamound Thanks for your PR! :tada: 

Just for the record, I think the `GZipResponder` class should **not** have a default value on `compresslevel`, as it's meant to be used by the `GZipMiddleware`. But as we expose the API of that module as is, I think it's better to not add breaking changes unnecessarily.

**Pull request merged**